### ISSUE TITLE
Allow updtr to skip certain dependencies?

### ISSUE THREAD

**Issue by bebraw** [peerigon/updtr#9]
There are certain projects where I prefer to use an older version of certain dependencies (Babel 5 related). Would it make sense to implement some kind of a mechanism to allow skipping these? A lazy way would be to do something like _.updtrignore_:

``` bash
babel-eslint
babel-loader
```

It simply wouldn't touch these files when running. Of course the con is that you end up having one more dotfile at your project. A decent alternative would be to push this information to _package.json_ and parse it from there.

**Comment by matthaias** [comment]
As you said, an extra dotfile will just add another dotfile in your project, what could end in a huge list of dotfiles. Also i am not sure if it is a good idea to add this to the `package.json`. 

What do you think about a `-e` or `--exclude` flag or something like this? 

Example:
```
$ updtr --exclude babel-eslint,babel-loader
```

**Comment by bebraw** [comment]
@matthaias `-e` or `--exclude` would work for me.

**Comment by ericclemmons** [comment]
Since this project is something you'd likely use intentionally (in a non-automated fashion like GreenKeeper), I suggested:
> A `--interactive` mode would be good. Each dep gets a prompt for `[(U)pdate/(i)gnore]`.
> – [@ericclemmons](https://twitter.com/ericclemmons/status/670654329611640832)

My projects have soooo many dependencies it'd be more of a chore to list them than to just let them all go through.

Although, I suppose with an `--exclude` option you'd want to put this into an `npm run` command.

**Comment by anselmh** [comment]
What about an `--interactive`/`-i` flag that lets you choose things interactively with `y`/`n`/`major`/`minor`/`patch` as options? Similar to what @ericclemmons proposes but more flexible.

This would be very useful for a dependency like jQuery 1.x where I don’t want to update to 2.x versions but only to minor versions.

That way you can select individually which version for which module you want.
I’d see this as an additional option to `--exclude`.

**Comment by bebraw** [comment]
What if the --interactive fiag generated a file based on the answers so
that it remembers them the next time you use it. Then you could just hit
return to go through them the next time you need to update. Just a thought.

**Comment by anselmh** [comment]
@bebraw That’d definitely be a cool thing. :)

**Comment by matthaias** [comment]
> This would be very useful for a dependency like jQuery 1.x where I don’t want to update to 2.x versions but only to minor versions.

For this you can use `--wanted` to update to specified version in your package.json. 

I think i can add the `--exclude`-option first. Would this help you already? First start with `--wanted` and then exclude the modules you don't want to update to the latest?

The interactive mode is going to be a harder feature in the current architecture i think.

**Comment by anselmh** [comment]
> I think i can add the --exclude-option first. Would this help you already? First start with --wanted and then exclude the modules you don't want to update to the latest?

Of course it will. interactive mode can of course be added later as an improvement, no worries – we all have limited time resources :)

**Comment by matthaias** [comment]
I added the option in the branch `feature/possibility-to-exclude-modules`. Can you test it please?

Example:
```
➜  updtr-test  npm i servus.js@0.0.1 --save
servus.js@0.0.1 node_modules/servus.js
➜  updtr-test  npm i xunit-file@0.0.1 --save
xunit-file@0.0.1 node_modules/xunit-file
➜  updtr-test  npm i erroz@0.0.1 --save
erroz@0.0.1 node_modules/erroz
➜  updtr-test  updtr --exclude servus.js,xunit-file
Found 1 outdated module

Starting to update your modules...
1/1 	 ● erroz 1.0.0 success

Finished
```

**Comment by matthaias** [comment]
Published as v0.1.6

**Comment by jhnns** [comment]
Imho an interactive mode would almost defeat the purpose of the module.

If you have to enter something for every dependency, it's almost like you'd bump the version manually in your `package.json` and run `npm test` afterwards.
